### DISCUSSION TITLE
Issue during RE training: pred_relation[1:pred_relation[0, 0, 0] + 1] list indices must be integers or slices, not tuple

### DISCUSSION THREAD

**Discussion by lamaeldo** [PaddlePaddle/PaddleOCR#12796] [Q&A]
Please provide the following information to quickly locate the problem

- System Environment：Debian 11
- Version：Paddle：2.5.1 (GPU)  PaddleOCR： 2.7.0.1 Related components：CUDA: 11.7 CUDNN: 8.2
- Command Code：`tools/train.py -c (path to my config file for RE training with LMv2)`
- Complete Error Message：
```
Traceback (most recent call last):
  File "/users/k2K/PaddleOCR/tools/train.py", line 227, in <module>
    main(config, device, logger, vdl_writer)
  File "/users/k2K/PaddleOCR/tools/train.py", line 198, in main
    program.train(config, train_dataloader, valid_dataloader, device, model,
  File "/users/k2K/PaddleOCR/tools/program.py", line 392, in train
    cur_metric = eval(
                 ^^^^^
  File "/users/k2K/PaddleOCR/tools/program.py", line 568, in eval
    post_result = post_process_class(preds, batch_numpy)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/users/k2K/PaddleOCR/ppocr/postprocess/vqa_token_re_layoutlm_postprocess.py", line 27, in __call__
    pred_relations = self.decode_pred(pred_relations)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/users/k2K/PaddleOCR/ppocr/postprocess/vqa_token_re_layoutlm_postprocess.py", line 62, in decode_pred
    pred_relation = pred_relation[1:pred_relation[0, 0, 0] + 1]
                                    ~~~~~~~~~~~~~^^^^^^^^^
TypeError: list indices must be integers or slices, not tuple
eval model::   0%|                                                                                                                                                         | 0/1 [00:02<?, ?it/s]
```

Description:
Hello, I am training RE models with PPOCR. I have a custom dataset, which works since it has allowed me train a decent model with VI_LayoutXLM . When i try to train a model with the LMV2 architecture (default config file, only tweaked the paths to my custom dataset). The training starts for the first few iterations but crashes at the first evaluation. I was at first surprised by this since all the code around the evaluation and the post-processing class (in KIE) makes no distinction between the different architectures, so I expected that if the evaluation worked with VI_XLM, it should work with LMV.
I dug a bit in the code, and it seems like it's not a simple syntax issue as the final error line would have one believe: the model output that is fed (`preds`) to the post processing class **does** contain a `pred_relations` element in the output of VI_XLM, in my case a tensor of shape `shape=[4, 263169, 7, 2]`. On the other hand, the model output with LMV2 contains a `pred_relations`, but it is empty apart from its structure: `'pred_relations': [[], [], [], []]`
I am guessing this means there is a bug in the output of LMV2 when used for RE? Or at least in the way its RE pairings are outputted or at least processed. I have seen #8995 but it did not help me.
I would be grateful for any help, since in my case, RE performance would be much improved by the use of visual features (which LMV2 supports as opposed to VI_XML)

**Comment by lamaeldo** [comment]
@tink2123 any ideas?

**Comment by Double97828** [comment]
any ideas?
